### ISSUE TITLE
Question: Can this be used to async-ify DataTable.Load()?

### ISSUE THREAD

**Issue by coderb** [Jehoel/AsyncDataAdapter#5]
hi.  can this library be used to eliminate the blocking call to `table.Load()` below in favor of somthing async?  thanks.
```
DbDataReader sqlDataReader =  await cmd.ExecuteReaderAsync(ct);
var table = new DataTable();
table.Load(sqlDataReader);
```

**Comment by daiplusplus** [comment]
Yes, that scenario is covered in the code-samples in the `README` file.

The sample uses `SqlAsyncDbDataAdapter .FillAsync( DataTable )` instead of `DataTable.Load` - however the two scenarios are roughly equivalent.

So the example code from the README is all you need:

```csharp
    using (SqlConnection conn = await CreateOpenConnectionAsync())
    using (SqlCommand cmd = conn.CreateCommand())
    {
        cmd.CommandText = "dbo.GetFast";
        cmd.CommandType = CommandType.StoredProcedure;
        cmd.Parameters.Add("@Number", SqlDbType.Int).Value = 100000;

        using(SqlAsyncDbDataAdapter adapter = cmd.CreateAsyncAdapter()) // `CreateAsyncAdapter` is an extension method.
        {
            DataTable singleTable = new DataTable();
            Int32 rows = await a.FillAsync(singleTable, cts.Token );
            return ds;
        }
    }
```

------------

[If you look at the implementation of `DataTable.Load`,](https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System.Data/System/Data/DataTable.cs#L4463) you'll see that it does this:

1. It instantiates a `adapter  = new Common.LoadAdapter();` (which abstracts-away the different `DbDataAdapter` subclasses like `SqlDataAdapter`, `OleDbDataReader`, etc.
2. it then calls `adapter.FillFromReader`, passing `this` into a new array as a parameter.

I adapted `FillFromReader` to `FillFromReaderAsync` in this project here: https://github.com/Jehoel/AsyncDataAdapter/blob/master/AsyncDataAdapter/Data/_DataAdapter/ProxyDbDataAdapter.FillAsync.cs

-------

## TL;DR:

Instead of `singleDataTable.Load( IDataReader )` use `await asyncDbDataAdapter.FillAsync( singleDataTable )`.

-------

## What about a `DataTable` async extension method `LoadAsync`?

Actually that's a good idea - I'll add it as a new issue right here: https://github.com/Jehoel/AsyncDataAdapter/issues/6

**Comment by coderb** [comment]
cool- i got it working with the example.  my old code uses a `SqlDataReader` and calls `table.Load` multiple times to get multiple tables from a stored procedure (stopping when `reader.IsClosed` is true).

i can't figure out how to get the additional results after the first table FillAsync.  i tried calling FillAsync twice in sequence but i believe i am incorrectly getting the first table back twice.

thanks.

**Comment by daiplusplus** [comment]
> @coderb i can't figure out how to get the additional results after the first table FillAsync

 You're using `SqlDataReader` and/or `SqlDataAdapter` incorrectly.

 If there are multiple resultsets then don't use the `DataTable` overload, use a single `DataSet`: the adapter will add new populated `DataTable` instances to the `DataSet` for each resultset.

```csharp
    using (SqlConnection conn = await CreateOpenConnectionAsync())
    using (SqlCommand cmd = conn.CreateCommand())
    {
        cmd.CommandText = "SELECT 1 AS R1C1; SELECT 2 AS R2C1, 'a' AS RC2; SELECT 3 AS R3C1, 0xDEADBEEF AS R3C2;";
        cmd.CommandType = CommandType.Text;

        using( SqlAsyncDbDataAdapter adapter = cmd.CreateAsyncAdapter() )
        {
            DataSet multipleTablesFromMultipleResultSets = new DataSet();
            _ = await a.FillAsync( multipleTablesFromMultipleResultSets, cts.Token );
            return multipleTablesFromMultipleResultSets;
        }
    }
```

If this library isn't doing that, then that's a bug - so send me instructions to reproduce it and I'll see if I can fix it.

**Comment by coderb** [comment]
@Jehoel my existing code iterates over the result sets which i process as they come back (and display in my u/i).  some result sets take longer than others so i need to be able to display them as they come rather than waiting for the complete data set.  is this possible with the library?  thanks.

**Comment by daiplusplus** [comment]
@coderb In that case I would _not_ use `SqlAsyncDbDataAdapter` at all and instead would fill each `DataTable` manually using the exposed methods in https://github.com/Jehoel/AsyncDataAdapter/blob/master/AsyncDataAdapter/Data/Core/FillAsync.cs - and after each `DataTable` is filled I'd give it to a consumer via `Progress<DataTable>`.
